### PR TITLE
HPCC-13714 Option for #Webservice(Fields()) to include unspecified fields

### PR DESCRIPTION
**Pull request by afishbeck**
The following syntax:
# WEBSERVICE(FIELDS('field1','field2','*'));

The addition of the name '*' would cause the form to show
first field1 and field2 and then all additional stored inputs
(unless noinput is specified for that stored).

Signed-off-by: Anthony Fishbeck anthony.fishbeck@lexisnexis.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@wangkx Please review

File: esp/services/ws_ecl/ws_ecl_wuinfo.cpp [hunk 1]
```diff
@@ -209,11 +209,39 @@ bool WsEclWuInfo::getWsResource(const char *name, StringBuffer &out)
...
+                        SCMStringBuffer s;
+                        IConstWUResult &var = vars->query();
+                        if (NotFound == fields.find(var.getResultName(s).str()))
+                        {
+                            SCMStringBuffer fieldSeq;
+                            var.getResultFieldOpt("sequence", fieldSeq);
+                            int seq = fieldSeq.length() ? atoi(fieldSeq.str()) : 0;
+                            if (seq < 0)
+                                seq = 0;
+                            seq = seq + i + 1;
```
Comment (wangkx) [inline esp/services/ws_ecl/ws_ecl_wuinfo.cpp]: I do not understand the logic here. Does the 'fieldSeq' give the offset from current location i? Can we assume that it returns continuous '1, 2, ..., n'?

**Comment by wangkx** [comment]
@afishbeck looks fine. Just one question.

File: esp/services/ws_ecl/ws_ecl_wuinfo.cpp (hunk 1 above)
Comment (afishbeck) [inline esp/services/ws_ecl/ws_ecl_wuinfo.cpp]: No the fieldSeq is being adapted to work a bit differently here.  When #WebService is not in use the fieldSeq is a relative position to all the other fields.  I.e. for managing over time you may use sparse numbering..   100, 150, 200, 300.  So I can easily insert a new entry somewhere.

What I am doing here allows you to put those in your #WebService field list, so I respect that numbering but make sure it fits where it belongs in our list.  Basically things inserted here in the list use fieldSeq only to determine there order relative to each other, but not fields named explicitly in the list.

So WebService(Fields('abc', *, def'));
anything added by '*' uses fieldSeq, but must be inserted between abc and def which do not use fieldSeq.

Comment (wangkx) [inline esp/services/ws_ecl/ws_ecl_wuinfo.cpp]: Ok

**Pull request merged**